lib/ts/fetch.ts

```diff
@@ -636,55 +621,133 @@ export async function getIdRefreshToken(tryRefresh: boolean): Promise<IdRefreshT
+        // in which case, we will not end up firing the SIGN_OUT on handle event.
+        expires = new Date(expiry).toUTCString();
+    }
+    const domain = AuthHttpRequest.config.sessionTokenFrontendDomain;
+    if (
+        domain === "localhost" ||
+        domain === WindowHandlerReference.getReferenceOrThrow().windowHandler.location.getHostName()
+    ) {
+        // since some browsers ignore cookies with domain set to localhost
+        // see https://github.com/supertokens/supertokens-website/issues/25
+        return CookieHandlerReference.getReferenceOrThrow().cookieHandler.setCookie(
+            `${name}=${value};expires=${expires};path=/;samesite=${
+                AuthHttpRequest.config.isInIframe ? "none;secure" : "lax"
+            }`
+        );
+    } else {
+        return CookieHandlerReference.getReferenceOrThrow().cookieHandler.setCookie(
+            `${name}=${value};expires=${expires};domain=${domain};path=/;samesite=${
+                AuthHttpRequest.config.isInIframe ? "none;secure" : "lax"
+            }`
+        );
+    }
+}
+
+export async function getToken(tokenType: TokenType) {
+    const name = getStorageNameForToken(tokenType);
+
+    return getFromCookies(name);
+}
+
+async function getFromCookies(name: string) {
+    let value = "; " + (await CookieHandlerReference.getReferenceOrThrow().cookieHandler.getCookie());
+    let parts = value.split("; " + name + "=");
+    if (parts.length >= 2) {
+        let last = parts.pop();
+        if (last !== undefined) {
+            return last.split(";").shift();
         }
-        if (
-            domain === "localhost" ||
-            domain === WindowHandlerReference.getReferenceOrThrow().windowHandler.location.getHostName()
-        ) {
-            // since some browsers ignore cookies with domain set to localhost
-            // see https://github.com/supertokens/supertokens-website/issues/25
-            await CookieHandlerReference.getReferenceOrThrow().cookieHandler.setCookie(
-                `${ID_REFRESH_TOKEN_NAME}=${cookieVal};expires=${expires};path=/;samesite=${
-                    AuthHttpRequest.config.isInIframe ? "none;secure" : "lax"
-                }`
-            );
-        } else {
-            await CookieHandlerReference.getReferenceOrThrow().cookieHandler.setCookie(
-                `${ID_REFRESH_TOKEN_NAME}=${cookieVal};expires=${expires};domain=${domain};path=/;samesite=${
-                    AuthHttpRequest.config.isInIframe ? "none;secure" : "lax"
-                }`
-            );
+    }
+    return undefined;
+}
+
+async function setTokenHeadersIfRequired(clonedHeaders: Headers, addRefreshToken: boolean = false) {
+    if (AuthHttpRequest.config.tokenTransferMethod === "header") {
+        logDebugMessage("setTokenHeadersIfRequired: adding ';header' to the rid header");
+        clonedHeaders.set("rid", clonedHeaders.get("rid") + ";header");
```

Comment:
- I think we had decided to extract this into another header, instead of appending to rid
- The default on the backend is header. So this condition should be about cookies